costs/es_request_parser.go

```diff
@@ -0,0 +1,128 @@
+//   Copyright 2017 MSolution.IO
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package costs
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"gopkg.in/olivere/elastic.v5"
+)
+
+// flattenSubAggregation allows to transform an array
+// of map to one flattened map.
+func flattenSubAggregation(in *interface{}) {
```

Comment:
This function does not return anything. Please either explicitly describe what this function modify, or return a value.
Functions that do not return and modify a parameter passed to it makes for quickly unmaintainable code.